airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/interpolated_paginator.py

```diff
@@ -6,15 +6,17 @@
 
 import requests
 from airbyte_cdk.sources.declarative.decoders.decoder import Decoder
+from airbyte_cdk.sources.declarative.decoders.json_decoder import JsonDecoder
 from airbyte_cdk.sources.declarative.interpolation.interpolated_mapping import InterpolatedMapping
 from airbyte_cdk.sources.declarative.interpolation.jinja import JinjaInterpolation
 from airbyte_cdk.sources.declarative.requesters.paginators.paginator import Paginator
+from airbyte_cdk.sources.declarative.types import Config
 
 
 class InterpolatedPaginator(Paginator):
-    def __init__(self, next_page_token_template: Mapping[str, str], decoder: Decoder, config):
+    def __init__(self, *, next_page_token_template: Mapping[str, str], config: Config, decoder: Optional[Decoder] = None):
```

Comment:
add type annotations and force the parameters to be passed by names to avoid accidentally mixing up the next_page_token_template and the config